src/system.cpp

```diff
@@ -2012,4 +2015,63 @@ bool System::uploadFirmwareURL(const char * url) {
     return true; // OK
 }
 
+// read command, e.g. read <deviceID> <type ID> [offset] [length] from console or API
+bool System::readCommand(const char * data) {
+    // extract <deviceID> <type ID> [offset] [length] from string
+    char * p;
+    char   value[10] = {0}; // null just in case
+
+    // make a copy so we can iterate
+    char data_args[EMS_MAX_TELEGRAM_LENGTH];
+    for (uint8_t i = 0; i < strlen(data); i++) {
+        data_args[i] = data[i];
+    }
+    data_args[strlen(data)] = '\0'; // make sure its terminated
```

Comment:
EMS_MAX_TELEGRAM_LENGTH seems not the logical., it's not a telegram.
the for.. copy can overflow if data is longer, why not using strlcpy?

Reply:
that is better indeed